### ISSUE TITLE
[TW-1703] When on-modify hook is installed, some messages print UUIDs instead of IDs

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1727]
_Robin Green on 2015-09-20T12:15:43Z says:_ 

 This happens regardless of whether the hook makes changes or not.

Example output:

{code}
$ task 68 annotate https://github.com/nidi3/raml-tester/issues/52
Annotating task 2173d9ba-6ad3-4c5b-8651-c490aed093bf 'Fix ! include ignored in resource type parameters'.
Annotated 1 task.
Project 'ST' is 33% complete (2 of 3 tasks remaining).
{code}

The task in this example is not done.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-09-20T12:15:43Z
Modified: 2016-02-02T15:06:30Z
```

**Comment by taskwarrior** [comment]
_Robin Green on 2015-10-18T11:50:10Z says:_ 

 In addition (I think this is probably caused by the same underlying bug) if a task is started that is recording as depending on another task, which has not yet been started, taskwarrior now prints "task 0" instead of the actual task number, in messages like this:

```
Starting task c825c3aa-ff7e-42bb-bc34-b1aa6b13aefd 'Use new integrated notmuch - deliver functionality'.
Task 0 is blocked by:
  62 Upgrade to Fedora 22
Started 1 task.
```

**Comment by taskwarrior** [comment]
_Tomas Babej on 2015-11-01T02:00:34Z says:_ 

 This may already be fixed in 2.5.1 branch by the recent identifier refactoring.
